### DISCUSSION TITLE
Ping needed to wake dead nodes but zwavejs2mqtt re-start won't

### DISCUSSION THREAD

**Discussion by raym777** [zwave-js/zwave-js-ui#1452] [General]
Hi guys. I originally posted this query on the the HA forum but someone suggested it would be more appropriate to raise it here.

I'm running the zwavejs2mqtt add-on in HA but with the MQTT gateway disabled (just using the console). Anyway, on a couple of occasions in recent weeks, my server went offline unexpectedly (due to a power failure). When it came back up, I noticed all my powered nodes were marked dead. Restarting the zwavejs2mqtt add-on didn't help. However, simply switching each node manually or sending a ping through the console immediately brought them back to life.

My question is, at system start-up, don’t all nodes get queried and if so, why didn’t it realised that the nodes were actually alive during this process? Is it possible to send a ping automatically during this startup process to check if nodes are really dead?

Appreciate any feedback on this one.

**Comment by robertsLando** [comment]
@AlCalzone

**Comment by AlCalzone** [comment]
> Is it possible to send a ping automatically during this startup process to check if nodes are really dead?

If you check the logs, this is exactly what happens for permanently-powered devices. Battery-powered devices are just assumed to be asleep.

**Reply by raym777** [reply]
Thanks for the reply. I wonder why I needed to manually send a ping to my permanently powered nodes then? Restarting alone did not detect the nodes as alive in my case.

**Reply by AlCalzone** [reply]
Probably best if you post a driver logfile of the startup.

**Reply by raym777** [reply]
Hi @AlCalzone, apologies but could you please help me understand the best way to enable and retrieve the logs? 

There appears to be 3 separate areas for enabling logs: 1) on the ZwaveJS2MQTT add-on Configuration page, 2) on the ZwaveJS2MQTT console under "general settings" and 3) on the HA Z-Wave JS Integration itself. 

The only option that appears to yield any related output when choosing a "debug" log level is option 2 (general settings) so I assume this is correct but, I'm only seeing the last part of the log in the UI. I've enabled the option to store logs to a file "store/zwavejs2mqtt.log" but I can't find this file anywhere on the filesystem.

Any help appreciated...

**Reply by robertsLando** [reply]
You can select logging in both general and zwave section of settings. If you don't see it is just because the 'enable logging' flag is disabled

**Reply by raym777** [reply]
Thanks @robertsLando - see stuff logged now in the UI but where is the raw (complete) file please?

**Reply by AlCalzone** [reply]
@raym777 See here: https://zwave-js.github.io/zwavejs2mqtt/#/troubleshooting/bug_report?id=driver-logs
To be sure: please post the **DRIVER** log, not the application log.

**Reply by raym777** [reply]
@AlCalzone Many thanks for the link. As it turns out, I'm unable to reproduce this issue now. Perhaps it needs a real power failure event to trigger some unique set of events - not sure. I'll keep an eye on it and post back my findings if and when it occurs again - if that's ok...
